Sorts/QuickSortMedianOfThree.java

```diff
@@ -0,0 +1,88 @@
+
+public class QuickSortMedianOfThree {
+
+    public void sort(int[] array, int leftIndex, int rightIndex){
+
+        if (leftIndex < rightIndex) {
+
+            medianOfThree(array, leftIndex, rightIndex);
+            changePivot(array, leftIndex, rightIndex);
+
+            int pivotIndex = partition(array, leftIndex, rightIndex);
+
+            sort(array, leftIndex, pivotIndex - 1);
+            sort(array, pivotIndex + 1, rightIndex);
+
+        }
+    }
+
+    private void medianOfThree(int[] array, int leftIndex, int rightIndex){
+
+        int shortter = leftIndex;
+        int mid = leftIndex + (rightIndex - leftIndex) / 2;
+        int bigger = rightIndex;
+
+        if (array[mid] < array[shortter]) {
+            swap(array, shortter, mid);
+        }
+        if (array[bigger] < array[shortter]) {
+            swap(array, shortter, bigger);
+        }
+        if (array[bigger] < array[mid]) {
+            swap(array, bigger, mid);
+        }
+
+    }
+
+    private void changePivot(int[] array, int leftIndex, int rightIndex) {
+        int pivotIndex = leftIndex + (rightIndex - leftIndex) / 2;
+        if (rightIndex - leftIndex >= 1) {
+            swap(array, pivotIndex, rightIndex - 1);
+        }
+
+    }
+
+    private int partition(int[] array, int leftIndex, int rightIndex) {
+
+        int i = leftIndex;
+        int j = leftIndex + 1;
+        int p = leftIndex;
+
+        while (j <= rightIndex) {
+
+            if (array[j] < array[p]) {
+                i = i + 1;
+                swap(array, i, j);
+            }
+            j = j + 1;
+        }
+
+        swap(array, i, p);
+        return i;
+    }
+
+    private void swap(int[] array, int i, int j) {
+        if(array == null)
```

Comment:
* Use curly braces.  
* This exception handling you can write to the methods above, too.